core/string/ustring.cpp

```diff
@@ -4364,6 +4364,17 @@ String String::property_name_encode() const {
 	return *this;
 }
 
+const String String::invalid_node_name_characters = ". : @ / \"";
+
+String String::validate_node_name() const {
+	Vector<String> chars = String::invalid_node_name_characters.split(" ");
```

Comment:
I know this is pre-existing code but I'm not sure if there's much gain having this as a static constant which then gets `split()` on each call. Compilers might be able to optimize this I guess? But we could just as well have `const String restricted_chars = ".:@/\"";` inside the function and iterate over the characters directly.

Reply:
Ah I see `invalid_node_name_characters` is used in SceneTreeEditor` to show a nice error message. Makes sense then I guess.

Reply:
I thought the same (I'm dubious that this would get memoized by the compiler) but my gut feel is that `validate_node_name` is probably not called frequently enough for the `split` to have a meaningful impact on performance in typical scenarios. If there are scenarios where very large numbers of nodes are getting created it'd be trivial to modify the error message case (which should be even more rare), but I wanted to keep changes limited to the original bug as much as practical.

Reply:
```suggestion
// Changes made to the set of invalid characters must also be reflected in the String documentation.
const String String::invalid_node_name_characters = ".:@/\"";

String String::validate_node_name() const {
	String invalid_char(&String::invalid_node_name_characters[0], 1);
	String replacement("");
	String name = this->replace(invalid_char, replacement);
	for (int i = 1; i < String::invalid_node_name_characters.length(); ++i) {
		invalid_char[0] = String::invalid_node_name_characters[i];
		name = name.replace(invalid_char, replacement);
	}

	return name;
}
```